go-apps/meep-metrics-engine/server/v2/metrics-engine.go

```diff
@@ -589,28 +602,82 @@ func mePostSeqQuery(w http.ResponseWriter, r *http.Request) {
 				continue
 			}
 			t := metricTime.Format("15:04:05.000")
+
+			if len(eventMetrics) > 0 {
+				eventMetric := eventMetrics[len(eventMetrics)-1]
+				eventMetricTime, err := time.Parse(time.RFC3339, eventMetric.Time.(string))
+				if err != nil {
+					log.Error("Failed to parse event time with error: ", err.Error())
+					continue
+				}
+				eventSdorg := ""
+				eventMermaid := ""
+				if eventMetricTime.Before(metricTime) {
+					log.Info("metricTime: ", t)
+					log.Info("eventMetricTime: ", eventMetricTime.Format("15:04:05.000"))
+					// Close previous mobility event group if necessary
+					if mobilityEventProcessed {
+						eventSdorg += "end\n"
+					}
+					mobilityEventProcessed = true
+
+					// Create group for mobility event
+					eventDescription := eventMetric.Description
+					eventDescription = strings.Replace(eventDescription, "[", "", -1)
+					eventDescription = strings.Replace(eventDescription, "]", "", -1)
+					eventSdorg += "\ngroup Mobility Event: " + eventDescription + "\n"
+					eventMermaid += "note over event: Mobility Event: " + eventDescription + "\n"
+					if params.Fields[0] == "mermaid" {
```

Comment:
Use the field constants defined in the metrics package.
NOTE: this comment applies to any other hardcoded field strings in this file.

Reply:
Addressed.